helpers/sql-obfuscation/lib/opentelemetry/helpers/sql_obfuscation.rb

```diff
@@ -127,6 +129,8 @@ def obfuscate_sql(sql, obfuscation_limit: 2000, adapter: :default)
 
       # @api private
       def truncate_statement(sql, regex, limit)
+        sql = sql.gsub(PREPENDED_COMMENT_REGEX, PLACEHOLDER) if sql.match?(PREPENDED_COMMENT_REGEX)
```

Comment:
The problem seems to be that `sql.index(regex)` returns `0` when a prepended comment is present, so our trick to truncate the message at the first match doesn't work, and the un-obfuscated query is preserved. 

We could ditch the query altogether and just return the truncation message by altering the condition on line 136: 
```diff 
-        return truncation_message unless first_match_index
+       return truncation_message unless first_match_index && first_match_index > 0
```

But then the user would get something like "SQL truncated (> 2000 characters)" rather than a snippet of the query until the first match.

There might be more performant ways to do this, I'd love to hear other ideas.

My benchmark is a little wacky, so I'll just share the results for now. I can post it if anyone is interested. 

Here are the current results:
| case | i/s |
| ---- | ---- |
|**with comment, altered return on line 136** |  295981.8 i/s |
|**with comment, current code** (this has an unobfuscated result) |  262156.7 i/s - 1.13x slower |
|**without comment, current code** |    92768.5 i/s - 3.19x slower |
|**without comment, altered return on line 136** |  87121.0 i/s - 3.40x slower |
|**without comment, with PREPENDED_COMMENT_REGEX checks** |   79018.5 i/s - 3.75x slower |
|**with comment, with PREPENDED_COMMENT_REGEX** |   66864.0 i/s - 4.43x slower |